### PR TITLE
adding spec to guard the invoke bundler behavior

### PR DESCRIPTION
**Pull request by ph**
Since the invoke_bundler changes a lot in the latest master changes It doesnt need refactor.
I have decided to add tests to guard the behavior of this method.

### CODE REVIEW

**Comment by ph** [comment]
Fixes https://github.com/elasticsearch/logstash/issues/2623

**Comment by colinsurprenant** [comment]
LGTM

**Comment by ph** [comment]
Why jenkins did not check this PR?

**Comment by mrsolo** [comment]
add to whitelist

**Comment by purbon** [comment]
LGTM, go for it :shipit: :shipit: :shipit: :shipit: :shipit:

**Comment by ph** [comment]
rebase make the tests failed looking at it.

**Comment by purbon** [comment]
Are your test failing because of https://github.com/elastic/logstash/issues/3018, all test are going to be red until we fix this.

**Comment by ph** [comment]
Not other one, I've changed the exception handling in the code.

**Comment by purbon** [comment]
Ok! :-)

**Comment by purbon** [comment]
Can we also make sure that https://github.com/elastic/logstash/pull/2752 goes in :-) is a very nice to have :rocket:

**Comment by ph** [comment]
@purbon review this one, checking the other one.

File: spec/lib/logstash/bundler_spec.rb [hunk 1]
```diff
@@ -27,4 +27,96 @@
       expect(exception.message).to eq("baz")
     end
   end
+  
+  context 'when invoking bundler' do
+    original_stderr = $stderr
```
Comment (purbon) [inline spec/lib/logstash/bundler_spec.rb]: makes sense to have this in to before block as the one you have in the after for restoring? might be just before line 40?

File: spec/lib/logstash/bundler_spec.rb [hunk 2]
```diff
@@ -27,4 +27,96 @@
...
+      expect(::Bundler.settings[:without]).to eq(options.fetch(:without, []).join(':'))
+
+      expect(ENV['GEM_PATH']).to eq(LogStash::Environment.logstash_gem_home)
+
+      $stderr = original_stderr
+    end
+
+    it 'should call Bundler::CLI.start with the correct arguments' do
+      expect(::Bundler::CLI).to receive(:start).with(bundler_args)
+      subject
```
Comment (purbon) [inline spec/lib/logstash/bundler_spec.rb]: this use of subject is strange to me. Why not just doing the call to invoke bundler?

Comment (ph) [inline spec/lib/logstash/bundler_spec.rb]: I can, subject is declared with the appropriate arguments.

Comment (purbon) [inline spec/lib/logstash/bundler_spec.rb]: Even if it's not wrong, somehow i read it wrong ... then there is another case when you call something like ``expect(subject)```

**Comment by purbon** [comment]
Left some minor style comments, for the rest LGTM, test passing, except for the error in #3018

File: spec/lib/logstash/bundler_spec.rb (hunk 2 above)
Comment (ph) [inline spec/lib/logstash/bundler_spec.rb]: ty I'll adjust them to make you happy :dancer:

**Pull request closed without merging**